### PR TITLE
PR #2258 with conflicts conservatively resolved

### PR DESCRIPTION
**Pull request by MichaIng**
@Mictronics
I rebased your PR and resolved conflicts in a way to prefer the changes that have been done to dev. There isn't much left anymore. I know you did some other changes not directly related to Python 3. Could you have a look and maybe re-add them here? Things I found interesting:
- Using `/%(camera_id)/stream` for getting the IP camera stream, which is at least something which seems required for `mjpg-streamer`, but not sure whether/how it was intended to work before.
- In handlers you switched from `videodevice` to `video_device` motion.conf setting... Ah now I see this is indeed correct for recent motion versions, so we should re-add this change: https://motion-project.github.io/motion_config.html
    **EDIT: Done!**

Please see whether there were other important changes you want to have re-added.

### CODE REVIEW

File: motioneye/extra/camera-1.conf.example [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+movie_filename %Y-%m-%d/%H-%M-%S
+movie_max_time 0
+movie_output off
+movie_passthrough off
+movie_codec mp4
+movie_quality 75
+on_event_start /home/dietpi/motioneye/motioneye/scripts/relayevent.sh "" start %t
+on_event_end /home/dietpi/motioneye/motioneye/scripts/relayevent.sh "" stop %t
+on_movie_end /home/dietpi/motioneye/motioneye/scripts/relayevent.sh "" movie_end %t %f
+on_picture_save /home/dietpi/motioneye/motioneye/scripts/relayevent.sh "" picture_save %t %f
```
Comment (MichaIng) [inline motioneye/extra/camera-1.conf.example:83]: This should be something more generic, e.g. `/etc/motioneye/scripts/relayevent.sh`?

**Comment by MichaIng** [comment]
About the motion settings: I think we need to support both variants, based on the motion version, i.e. the old names for motion <=4.3 and the new variants for motion >=4.4. Is there a good way to check for the motion version?

File: motioneye/mediafiles.py [hunk 2]
```diff
@@ -742,6 +756,9 @@ def poll_movie_process(pictures):
                 else:
                     raise
 
+            if output is None:
+                output = ''
+
```
Comment (cclauss) [inline motioneye/mediafiles.py:761]:
```suggestion
            output = output or ''
```
All falsey values (False, 0, (), {}, [], etc.) are just as risky as None.

**Review by cclauss — APPROVED** [review]
RSLGMT

**Review by cclauss — APPROVED** [review]
RSLGTM

**Comment by MichaIng** [comment]
I reverted the motion config key changes. We should do this properly in a dedicated PR where we handle support for both versions. With the old/current keys we support motion up to v4.3, hence all versions shipped by distros, while we do not support latest upstream v4.4. The other changes are however pretty fine. Compared to #2258 I did some minor change to `extra/` file names and paths to keep them as they were before. Similarly, if possible, we should keep the original behaviour that these extra files (default configs, service) are installed to `/usr/local/share/motioneye` when the module is installed as root, so that generally non-required changes are minimal.

**Comment by MichaIng** [comment]
I found the original PRs of two of the major changes here. I'd prefer to merge those (done with one already). The remaining change here is the added sample motion configs, but I wonder what the benefit of them is, since motionEye creates default motion configs internally, and this hence would require doubled code changes, and hardcodes the configs for specific motion versions.

**Comment by MichaIng** [comment]
Marking this as closed in favour if the original PRs. Whether or not default motion configs are wanted, should be discussed in a separate issue/PR.

**Pull request closed without merging**